torch/package/_custom_import_pickler.py

```diff
@@ -54,6 +55,11 @@ def save_global(self, obj, name=None):
         lastname = name.rpartition('.')[2]
         if parent is module:
             name = lastname
+
```

Comment:
I think the right place to demangle is in `whichmodule`-> it should just remove the demangled name. Then `import_module` will resolve it to the first namespace which has the unmangled thing provided. The error reporting in the pickler should be improved so that when something doesn't match it reports that one was loaded from a package when it occurs.